nightly_upload.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+# nightly_upload: checks remote file for changes and uploads new version if necessary
+# Usage:
+# set scp(ssh) server with user and path
+# set ssh port
+# create env-variable with private key: $ssh_server_key
+# call script: ./nightly_upload.sh -filename
+
+repo_name=$(basename `git rev-parse --show-toplevel`)
+
+ssh="drupal@www.nitrokey.com:~/www/nitrokey.com/files/ci/"$repo_name'/' #-P 2215 drupal@www.nitrokey.com:~/www/nitrokey.com/files/ci/pynitro>
+port=2215
+build_file="$1"
+new_sha=$build_file"-sha256sum.txt"
+
+old_sha="old"-$new_sha
+date_now=$(date "+%F-%H-%M-%S")
+echo $repo_name
+
+# create ssh-key file
+echo "-----BEGIN OPENSSH PRIVATE KEY-----" > ~/server_key
+echo "$ssh_server_key" >> ~/server_key
+echo "-----END OPENSSH PRIVATE KEY-----" >> ~/server_key
+
+chmod 600 ~/server_key
+echo key-var
+echo $ssh_server_key
+echo key_file
+cat ~/server_key
+
+sha256sum $build_file > $new_sha
+scp -o StrictHostKeyChecking=accept-new -P $port -i ~/server_key $ssh$new_sha $old_sha
+if cmp -s $new_sha $old_sha; then
+    echo Build is unchanged
+else
+    scp -o StrictHostKeyChecking=accept-new -P $port -i ~/server_key $new_sha $ssh$new_sha
+    scp -o StrictHostKeyChecking=accept-new -P $port -i ~/server_key $build_file $ssh"nightly_build-"$date_now$build_file
```

Comment:
`scp` is outdated, please do not use it and replace it with `rsync`

> According to OpenSSH developers in April 2019, SCP is outdated, inflexible and not readily fixed; they recommend the use of more modern protocols like sftp and rsync for file transfer.[3]

- https://en.wikipedia.org/wiki/Secure_copy_protocol
- https://www.openssh.com/txt/release-8.0